### PR TITLE
Azure Monitor Exporter - Update SampleRate in Request/Dependency telemetry

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
- Update `SampleRate` in Request/Dependency telemetry if `Activity` has `sampleRate` attribute. 
- Added Tests and updated demo app

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Azure.Monitor.OpenTelemetry.Exporter.csproj [hunk 1]
```diff
@@ -13,6 +13,7 @@
     <PackageReference Include="System.Text.Json" />
     <PackageReference Include="OpenTelemetry" />
     <PackageReference Include="OpenTelemetry.Extensions.PersistentStorage" />
+    <PackageReference Include="OpenTelemetry.Extensions.AzureMonitor" />
```
Comment (cijothomas) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Azure.Monitor.OpenTelemetry.Exporter.csproj:16]: oops! do we need this! why would exporter depend on a Sampler !

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorExporterTraceExtensions.cs [hunk 2]
```diff
@@ -31,6 +32,8 @@ public static TracerProviderBuilder AddAzureMonitorTraceExporter(this TracerProv
             // TODO: provide a way to turn off statsbeat
             // Statsbeat.InitializeAttachStatsbeat(options.ConnectionString);
 
+            builder.SetSampler(new ApplicationInsightsSampler(options.SamplingRatio));
```
Comment (cijothomas) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorExporterTraceExtensions.cs:35]: I think we should just document people to add the sampler themselves. Exporter should have no business setting sampler for user.

File: eng/Packages.Data.props [hunk 3]
```diff
@@ -118,6 +118,7 @@
     <!-- OpenTelemetry dependency approved for Azure.Monitor.OpenTelemetry.Exporter package only -->
     <PackageReference Update="OpenTelemetry" Version="1.3.1"  Condition="'$(MSBuildProjectName)' == 'Azure.Monitor.OpenTelemetry.Exporter'" />
     <PackageReference Update="OpenTelemetry.Extensions.PersistentStorage" Version="1.0.0-beta.1" Condition="'$(MSBuildProjectName)' == 'Azure.Monitor.OpenTelemetry.Exporter'" />
+	<PackageReference Update="OpenTelemetry.Extensions.AzureMonitor" Version="1.0.0-beta.2" Condition="'$(MSBuildProjectName)' == 'Azure.Monitor.OpenTelemetry.Exporter'" />
```
Comment (weshaggard) [inline eng/Packages.Data.props:121]:
```suggestion
    <PackageReference Update="OpenTelemetry.Extensions.AzureMonitor" Version="1.0.0-beta.2" Condition="'$(MSBuildProjectName)' == 'Azure.Monitor.OpenTelemetry.Exporter'" />
```

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/TelemetryItem.cs [hunk 4]
```diff
@@ -43,6 +43,10 @@ internal partial class TelemetryItem
             }
 
             SetResourceSdkVersionAndIkey(roleName, roleInstance, instrumentationKey);
+            if (AzMonList.GetTagValue(ref monitorTags.MappedTags, "sampleRate") is float sampleRate)
```
Comment (cijothomas) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/TelemetryItem.cs:46]: make sure we have fall back. i.e we need to anticipate this tag not being present, and then don't send sample rate.

Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/TelemetryItem.cs:46]: We do have a fall back in searializer.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Azure.Monitor.OpenTelemetry.Exporter.csproj (hunk 1 above)
Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Azure.Monitor.OpenTelemetry.Exporter.csproj]: It's only a part of tests now.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/SampleRateTests.cs [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    {
+        private const string ActivitySourceName = "SampleRateTests";
+        private const string ActivityName = "TestActivity";
+
+        static SampleRateTests()
+        {
+            Activity.DefaultIdFormat = ActivityIdFormat.W3C;
+            Activity.ForceDefaultIdFormat = true;
+
+            var listener = new ActivityListener
```
Comment (cijothomas) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/SampleRateTests.cs:28]: its a bit odd that we have to setup own listeners? Can we use OTel SDK TracerProvider instead with AlwaysOnSampler?

Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/SampleRateTests.cs:28]: All our tests follow this model, we might need to refactor all of the tests in a different task.

Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/SampleRateTests.cs:28]: These tests were based on old OTel approach and needs an upgrade.

**Comment by rajkumar-rangaraj** [comment]
@vishweshbankwar As we discussed, added zero sampling case where no telemetries are created.

**Review by vishweshbankwar — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]
Please add changelog, and update PR description to reflect the actual changes.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/Azure.Monitor.OpenTelemetry.Exporter.Tests.csproj [hunk 6]
```diff
@@ -4,11 +4,16 @@
...
+  <ItemGroup>
+    <Compile Include="..\Integration.Tests\TestFramework\MockTransmitter.cs" Link="MockTransmitter.cs" />
+  </ItemGroup>
+
   <ItemGroup>
     <PackageReference Include="xunit" />
     <PackageReference Include="xunit.runner.visualstudio" />
     <PackageReference Include="Microsoft.NET.Test.Sdk" />
     <PackageReference Include="Moq" />
+    <PackageReference Include="OpenTelemetry.Extensions.AzureMonitor" VersionOverride="1.0.0-beta.2" />
```
Comment (TimothyMothra) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/Azure.Monitor.OpenTelemetry.Exporter.Tests.csproj:16]: does the test project need this explicit reference?
i would expect this to come from the Exporter project

**Review by TimothyMothra — APPROVED** [review]

**Pull request merged**